src/sagemaker/jumpstart/model.py

```diff
@@ -866,92 +874,94 @@ def register_deploy_wrapper(*args, **kwargs):
+    def _retrieve_selected_deployment_config(
+        self, config_name: str, instance_type: str
+    ) -> Optional[DeploymentConfigMetadata]:
+        """Retrieve the deployment config to apply to `This` model.
 
         Args:
             config_name (str): The name of the deployment config to retrieve.
+            instance_type (str): The instance type of the deployment config to retrieve.
         Returns:
             Optional[Dict[str, Any]]: The retrieved deployment config.
         """
         if config_name is None:
             return None
 
-        for deployment_config in self._deployment_configs:
-            if deployment_config.get("DeploymentConfigName") == config_name:
+        for deployment_config in self._get_deployment_configs(config_name, instance_type):
+            if deployment_config.deployment_config_name == config_name:
                 return deployment_config
         return None
 
-    def _convert_to_deployment_config_metadata(
-        self, config_name: str, metadata_config: JumpStartMetadataConfig
-    ) -> Dict[str, Any]:
-        """Retrieve deployment config for config name.
+    @lru_cache
+    def _get_deployment_configs(
+        self, selected_config_name: str, selected_instance_type: str
+    ) -> List[DeploymentConfigMetadata]:
+        """Retrieve deployment configs metadata.
 
         Args:
-            config_name (str): Name of deployment config.
-            metadata_config (JumpStartMetadataConfig): Metadata config for deployment config.
-        Returns:
-            A deployment metadata config for config name (dict[str, Any]).
+            selected_config_name (str): The name of the selected deployment config.
+            selected_instance_type (str): The selected instance type.
         """
-        default_inference_instance_type = metadata_config.resolved_config.get(
-            "default_inference_instance_type"
-        )
-
-        benchmark_metrics = (
-            metadata_config.benchmark_metrics.get(default_inference_instance_type)
-            if metadata_config.benchmark_metrics is not None
-            else None
-        )
-
-        should_fetch_instance_rate_metric = True
-        if benchmark_metrics is not None:
-            for benchmark_metric in benchmark_metrics:
-                if benchmark_metric.name.lower() == "instance rate":
-                    should_fetch_instance_rate_metric = False
-                    break
-
-        if should_fetch_instance_rate_metric:
-            instance_rate = get_instance_rate_per_hour(
-                instance_type=default_inference_instance_type, region=self.region
+        deployment_configs = []
+        if self._metadata_configs is None:
+            return deployment_configs
+
+        err = None
+        for config_name, metadata_config in self._metadata_configs.items():
+            if err is None or "is not authorized to perform: pricing:GetProducts" not in err:
```

Comment:
Is this only permission required?

Reply:
Not quite following the logic here. Say in the first iteration err is assigned with no pricing:GetProducts permission exception, is there a need to continue the rest of this function?

Reply:
Discussed offline, cleaner logic will be applied in following prs.

Reply:
We continue the rest but the only thing we skip is `add_instance_rate_stats_to_benchmark_metrics` if err contains `is not authorized to perform: pricing:GetProducts`

I'll clean it up to make it clear.